service/history/handler.go

```diff
@@ -1140,6 +1140,8 @@ func (h *Handler) updateErrorMetric(scope int, err error) {
 		h.metricsClient.IncCounter(scope, metrics.CadenceErrRetryTaskCounter)
 	case *gen.ServiceBusyError:
 		h.metricsClient.IncCounter(scope, metrics.CadenceErrServiceBusyCounter)
+	case *persistence.CurrentWorkflowConditionFailedError:
```

Comment:
Persistence errors are not defined on the service API.  These needs to be wrapped with an InternalServiceError or something.